src/Articulate/MetaWeblog/ArticulateMetaWeblogProvider.cs

```diff
@@ -0,0 +1,512 @@
+            }
+            if (!post.mt_excerpt.IsNullOrWhiteSpace())
+            {
+                content.SetInvariantOrDefaultCultureValue("excerpt", post.mt_excerpt, contentType, _localizationService);
+            }
+
+            // TODO: Investigate why this not supported ?!
+            //if (post.AllowComments == 1)
+            //{
+            //    content.SetInvariantOrDefaultCultureValue("enableComments", 1, contentType, _localizationService);
+            //}
+            //else if (post.AllowComments == 2)
+            //{
+            //    content.SetInvariantOrDefaultCultureValue("enableComments", 0, contentType, _localizationService);
+            //}
+
+            content.AssignInvariantOrDefaultCultureTags("categories", post.categories, contentType, _localizationService, _dataTypeService, _propertyEditors, _jsonSerializer);
+            var tags = post.mt_keywords.Split(new[] { ',' }, StringSplitOptions.RemoveEmptyEntries).Distinct().ToArray();
+
+            content.AssignInvariantOrDefaultCultureTags("tags", tags, contentType, _localizationService, _dataTypeService, _propertyEditors, _jsonSerializer);
+
+            if (publish)
+            {
+                if (post.dateCreated != DateTime.MinValue)
+                {
+                    content.SetInvariantOrDefaultCultureValue("publishedDate", post.dateCreated, contentType, _localizationService);
+                }
+
+                _contentService.SaveAndPublish(content, userId: user.Id);
+            }
+            else
+            {
+                _contentService.Save(content, user.Id);
+            }
+        }
+
+        /// <summary>
+        /// There are so many variants of Metaweblog API so I've just included as many properties, custom ones, etc... that i can find
+        /// </summary>
+        /// <param name="post"></param>
+        /// <returns></returns>
+        /// <remarks>
+        /// http://msdn.microsoft.com/en-us/library/bb463260.aspx
+        /// http://xmlrpc.scripting.com/metaWeblogApi.html
+        /// http://cyber.law.harvard.edu/rss/rss.html#hrelementsOfLtitemgt
+        /// http://codex.wordpress.org/XML-RPC_MetaWeblog_API
+        /// https://blogengine.codeplex.com/SourceControl/latest#BlogEngine/BlogEngine.Core/API/MetaWeblog/MetaWeblogHandler.cs
+        /// </remarks>
+        private Post FromPost(PostModel post) => new Post
+        {
+            categories = post.Categories.ToArray(),
+            description = post.Body.ToString(),
+            dateCreated = post.PublishedDate != default(DateTime) ? post.PublishedDate : post.UpdateDate,
+            postid = post.Id.ToString(CultureInfo.InvariantCulture),
+            wp_slug = post.Url(),
+            mt_excerpt = post.Excerpt,
+            mt_keywords = string.Join(",", post.Tags.ToArray()),
+            title = post.Name
+
+            // TODO: AllowComments & Author
```

Comment:
new TODO or old TODO?

Reply:
Will resolve, comments will be fixed when the https://github.com/shawnwildermuth/MetaWeblog/pull/16 is merged, we don'tneed to worry about author.